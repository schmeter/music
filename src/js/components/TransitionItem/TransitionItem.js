import React from 'react';
import PropTypes from 'prop-types';
import { TransitionGroup, CSSTransition } from 'react-transition-group';

class TransitionItem extends React.Component {
    render() {
        const { children } = this.props;
        const timeout = 300;
        const items = React.Children.toArray(children).map((item, index) => (
            <CSSTransition
                key={index}
                // in
                appear
                classNames="fade"
                timeout={timeout}
            >
                <div className="transition-wrapper">
                    {item}
                </div>
            </CSSTransition>
        ));

        return <TransitionGroup>{items}</TransitionGroup>;
    }
}

TransitionItem.propTypes = {
    children: PropTypes.node.isRequired
};

export default TransitionItem;
