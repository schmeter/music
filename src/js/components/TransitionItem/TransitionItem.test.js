import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TransitionItem from './TransitionItem';

describe('TransitionItem', () => {
  it('renders correctly', () => {
    const component = shallow(<TransitionItem>test</TransitionItem>);

    expect(toJson(component)).toMatchSnapshot();
  });
});
