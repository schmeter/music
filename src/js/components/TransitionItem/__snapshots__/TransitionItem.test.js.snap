// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransitionItem renders correctly 1`] = `
<TransitionGroup
  childFactory={[Function]}
  component="div"
>
  <CSSTransition
    appear={true}
    classNames="fade"
    key="0"
    timeout={300}
  >
    <div
      className="transition-wrapper"
    >
      test
    </div>
  </CSSTransition>
</TransitionGroup>
`;
